File: library_management/library_management/report/person_doc_script_report/person_doc_script_report.py
# Copyright (c) 2025, XYZ and contributors
# For license information, please see license.txt

import frappe

# def execute(filters=None):
#     # Default filters if none are provided
#     if filters is None:
#         filters = {}

#     # Query with the provided filters
#     data = frappe.get_all('Person', fields=["first_name", "last_name", "age"], filters=filters)
    
#     # Return data to be displayed in the report
#     return data
def execute(filters=None):
	# columns, data = [], []
	columns = [
        {"label": "First Name", "fieldname": "first_name", "fieldtype": "Data", "width": 150},
        {"label": "Last Name", "fieldname": "last_name", "fieldtype": "Data", "width": 150},
        # {"label": "Age", "fieldname": "age", "fieldtype": "Int", "width": 80},
    ]
	data = frappe.get_all("Person", fields=["first_name", "last_name"],filters=filters)

	return columns, data
    

File: library_management/library_management/report/person_doc_script_report/person_doc_script_report.js
// Copyright (c) 2025, XYZ and contributors
// For license information, please see license.txt

// frappe.query_reports["Person Doc Script REPORT"] = {
// 	"filters": [

// 	]
// };

// frappe.query_reports["Person Doc Script REPORT"] = {
//     "filters": [
//         {
//             "fieldname": "first_name",
//             "label": "First Name",
//             "fieldtype": "Data",
//             "reqd": 0
//         },
//         {
//             "fieldname": "age",
//             "label": "Age",
//             "fieldtype": "Int",
//             "default": 0,
//             "reqd": 0
//         }
//     ]
// };

frappe.query_reports['Person Doc Script REPORT'] = {
    filters: [
    {
    fieldname: 'first_name',
    label: __('First Name'),
    fieldtype: 'Data',
    options: '',
    default: ''
    },
    ]
   }

